### PR TITLE
HS-561: Add snmp collector

### PR DESCRIPTION
**Pull request by cgorantla**
## Description
Add Snmp Collector.
Collect SNMP Node metrics.

## Jira link(s)
- https://issues.opennms.org/browse/HS-561

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: inventory/src/main/java/org/opennms/horizon/inventory/service/taskset/CollectorTaskSetService.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.opennms.taskset.service.api.TaskSetPublisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CollectorTaskSetService {
+
+    private static final Logger log = LoggerFactory.getLogger(MonitorTaskSetService.class);
```
Comment (thomasbigger584) [inline inventory/src/main/java/org/opennms/horizon/inventory/service/taskset/CollectorTaskSetService.java:51]:
```suggestion
    private static final Logger log = LoggerFactory.getLogger(CollectorTaskSetService.class);
```

File: minion/minion-taskset/taskset-worker/src/main/java/org/opennms/horizon/minion/taskset/worker/impl/TaskExecutorLocalCollectorServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class TaskExecutorLocalCollectorServiceImpl implements TaskExecutorLocalService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TaskExecutorLocalMonitorServiceImpl.class);
```
Comment (thomasbigger584) [inline minion/minion-taskset/taskset-worker/src/main/java/org/opennms/horizon/minion/taskset/worker/impl/TaskExecutorLocalCollectorServiceImpl.java:49]:
```suggestion
    private static final Logger LOG = LoggerFactory.getLogger(TaskExecutorLocalCollectorServiceImpl.class);
```

File: shared-lib/snmp/snmp-api/src/main/java/org/opennms/horizon/shared/snmp/SnmpNodeTracker.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+
+public class SnmpNodeTracker extends AggregateTracker {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Mib2InterfacesTracker.class);
```
Comment (thomasbigger584) [inline shared-lib/snmp/snmp-api/src/main/java/org/opennms/horizon/shared/snmp/SnmpNodeTracker.java:40]:
```suggestion
    private static final Logger LOG = LoggerFactory.getLogger(SnmpNodeTracker.class);
```

File: minion/plugins/plugins-api/src/main/java/org/opennms/horizon/minion/plugin/api/CollectionSet.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.horizon.minion.plugin.api;
+
+import com.google.protobuf.Message;
+
+public interface CollectionSet {
+
+    Message getResults();
```
Comment (j-white) [inline minion/plugins/plugins-api/src/main/java/org/opennms/horizon/minion/plugin/api/CollectionSet.java:35]: Is `Message` the right type here?

Comment (cgorantla) [reply minion/plugins/plugins-api/src/main/java/org/opennms/horizon/minion/plugin/api/CollectionSet.java:35]: This CollectionSet interface could be common to all the collectors, The results are a different protobuf message for each collector.  using `google.protobuf.Any` for the results.

**Review by thomasbigger584 — APPROVED** [review]
lgtm

**Pull request merged**